### DISCUSSION TITLE
blog/setup-vscode-to-run-debug-c-cpp-code

### DISCUSSION THREAD

**Discussion by giscus** [GorvGoyl/Personal-Site-Gourav.io#35] [Announcements]
Discussion related to the article:

**Title:** By the end of this short guide, you’d be able to run, debug, and get IntelliSense for C/C++ files in VSCode
**URL:** https://gourav.io/blog/setup-vscode-to-run-debug-c-cpp-code

**Comment by ranjithmkumar1992** [comment]
Could you provide the steps for Mac M1?

**Reply by CaptainProfit** [reply]
ty

**Comment by dvogureckiy99** [comment]
Dude please help me, after I try compile< I have got this error massage:
The terminal process failed to launch: Starting directory (cwd) "C:\MinGW64\bin" does not exist

**Reply by GorvGoyl** [reply]
make sure the folder C:\MinGW64\bin is there in your system

**Comment by SyntaxError-Natsu** [comment]
I am getting an error that g++.exe is not found and configured the task.
I am configuring again and I have g++.exe in the bin folder but not debugging.

**Comment by AdmanAbid** [comment]
I did all of these and the debugger is running but the code is not interactive anymore. The outputs aren't showing in the terminal or external command prompt and I can't give any inputs to the code either. How can I configure so that the code runs in terminal while debugging?

Edit: When I set ExternalConsole to true the code runs properly with inputs and outputs but in an external command prompt, which is very hard to manage. Setting ExternalConsole to false gives me the problems where it doesn't use inputs or outputs.
